python/setup.py

```diff
@@ -0,0 +1,180 @@
+        # package root and we need to copy the jars and scripts which are up above the python root.
+        if getattr(os, "symlink", None) is not None:
+            os.symlink(JARS_PATH, JARS_TARGET)
+            os.symlink(SCRIPTS_PATH, SCRIPTS_TARGET)
+            os.symlink(EXAMPLES_PATH, EXAMPLES_TARGET)
+        else:
+            # For windows fall back to the slower copytree
+            copytree(JARS_PATH, JARS_TARGET)
+            copytree(SCRIPTS_PATH, SCRIPTS_TARGET)
+            copytree(EXAMPLES_PATH, EXAMPLES_TARGET)
+    else:
+        # If we are not inside of SPARK_HOME verify we have the required symlink farm
+        if not os.path.exists(JARS_TARGET):
+            print("To build packaging must be in the python directory under the SPARK_HOME.",
+                  file=sys.stderr)
+        # We copy the shell script to be under pyspark/python/pyspark so that the launcher scripts
+        # find it where expected. The rest of the files aren't copied because they are accessed
+        # using Python imports instead which will be resolved correctly.
+        try:
+            os.makedirs("pyspark/python/pyspark")
+        except OSError:
+            # Don't worry if the directory already exists.
+            True
+        copyfile("pyspark/shell.py", "pyspark/python/pyspark/shell.py")
+
+    if not os.path.isdir(SCRIPTS_TARGET):
+        print("You must first create a source dist and install that source dist.", file=sys.stderr)
+        exit(-1)
+
+    # Scripts directive requires a list of each script path and does not take wild cards.
+    script_names = os.listdir(SCRIPTS_TARGET)
+    scripts = map(lambda script: os.path.join(SCRIPTS_TARGET, script), script_names)
+    # We add find_spark_home.py to the bin directory we install so that pip installed PySpark
+    # will search for SPARK_HOME with Python.
+    scripts.append("pyspark/find_spark_home.py")
+
+    # Parse the README markdown file into rst for PyPI
+    long_description = "!!!!! missing pandoc do not upload to PyPI !!!!"
+    try:
+        import pypandoc
+        long_description = pypandoc.convert('README.md', 'rst')
+    except ImportError:
+        print("Could not import pypandoc - required to package PySpark", file=sys.stderr)
+
+    setup(
+        name='pyspark',
+        version=VERSION,
+        description='Apache Spark Python API',
+        long_description=long_description,
+        author='Spark Developers',
+        author_email='dev@spark.apache.org',
+        url='https://github.com/apache/spark/tree/master/python',
+        packages=['pyspark',
+                  'pyspark.mllib',
+                  'pyspark.ml',
+                  'pyspark.sql',
+                  'pyspark.streaming',
+                  'pyspark.bin',
+                  'pyspark.jars',
+                  'pyspark.python.lib',
+                  'pyspark.examples.src.main.python'],
```

Comment:
I'd rather not automate this - I think its more likely to go wrong than right. We also need some custom magic for the Python libs and the examples - so it would be partially manual anyways.